### PR TITLE
Spreadsheet plugin docs

### PR DESCRIPTION
**Pull request by f1ames**
* [x] Dev guides
* [x] Features overview
* [x] Sample

Since this PR adds `spreadsheet` plugin too, all files in `docs/sdk/examples/assets/plugins/spreadsheet/` are the plugin dist files and should be skipped during review.

### CODE REVIEW

File: docs/features/spreadsheets/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## Sorting
+
+Sorting is a must to efficiently analyze and present any tabular data, thus spreadsheet plugin allows for sorting data based on single or multiple columns:
+
+{@img assets/img/spreadsheets3.png Spreadsheet plugin widget with multisort}
+
+## Basic styles support
+
+While spreadsheets are mostly about data, its presentation is also very important. Basic styles like bold or italics comes handy when trying to highlight or make more visible some parts of data or single spreadsheet cells:
```
Comment (jacekbogdanski) [inline docs/features/spreadsheets/README.md]: There is also justify and removeformat support but not sure if it's worth pointing this out / treat as overall basicstyles support :thinking:

**Comment by Comandeer** [comment]
I've added info about error code for missing/invalid license key.

File: docs/features/spreadsheets/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+## Data types, formats and validation
+
+Data is the essence of every spreadsheet or data-grid element. This means it requires correct typing so for example formulas and conditional formatting knows how to handle it. Spreadsheet plugin provides few data types like Text, Number, Date, Time, Password and various formats so visually it can be presented in various ways. Additionally, each data type has a strict validation so it is clearly visible when something is not right with the data:
+
+{@img assets/img/spreadsheets5.png Spreadsheet plugin with Cells Type and Format dialog}
+
+## Formulas
+
+Formulas provide a way to calculate many different things and automate data processing in any spreadsheet. Just type `=` and the rest of the formula, easy as one two three!
```
[the remark this answers is not in the document]
Comment (f1ames) [reply docs/features/spreadsheets/README.md]: Makes sense 👍 Copied and linked :D

File: docs/features/spreadsheets/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+{@img assets/img/spreadsheets6.png Spreadsheet plugin with formula preview}
+
+## Conditional formatting
+
+Need to add some colors to your data? Conditional formatting allows to format data (any cell, column or entire table) based on its content. You may mark cells red when they are empty, make them green when theri value is above any threshold or blue if the cell value contains specific text.
```
[the remark this answers is not in the document]
Comment (f1ames) [reply docs/features/spreadsheets/README.md]: Yep, I have the similar feeling, it sometimes hard to write it the other way (non-personal). I will try to rephrase.

[the remark this answers is not in the document]
Comment (f1ames) [reply docs/features/spreadsheets/README.md]: There is more than I have expected :D

[the remark this answers is not in the document]
Comment (f1ames) [reply docs/features/spreadsheets/README.md]: I see the other docs also use this form, I will still consult with @AnnaTomanek if that's ok. cc @AnnaTomanek

File: docs/features/spreadsheets/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+{@img assets/img/spreadsheets5.png Spreadsheet plugin with Cells Type and Format dialog}
+
+## Formulas
+
+Formulas provide a way to calculate many different things and automate data processing in any spreadsheet. Just type `=` and the rest of the formula, easy as one two three!
+
+{@img assets/img/spreadsheets6.png Spreadsheet plugin with formula preview}
+
+## Conditional formatting
```
[the remark this answers is not in the document]
Comment (f1ames) [reply docs/features/spreadsheets/README.md]: I was also thinking about it but don't have any nice idea for now. Maybe adding such formatting in sample could be enough?

**Comment by f1ames** [comment]
> I'm wondering if it makes sense to write some guide about overall Handsontable features - like visual selection, copying values by selection etc. This guide mostly targets integration features, but we are selling this product as a whole, so maybe there should be also some copy&paste from HOT tutorials?

@jacekbogdanski that would make sense, but there is a lot o features. Which do you think would be most valuable to mention (e.g. I assume copy/paste is obvious so could be skipped - but copying visual selection and pasting outside as another instance is more interesting)?

File: docs/features/spreadsheets/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnnaTomanek) [reply docs/features/spreadsheets/README.md]:
```suggestion
Do you need to add some colors to your data? Conditional formatting allows formatting data (any cell, column or entire table) based on its content. You can mark cells red when they are empty, make them green when their value is above any threshold or blue if the cell value contains a specific text.
```

We actually prefer the "you" form to "we" form as it's more engaging and less patronizing; it's not "our software", it becomes more of "your implementation", if you get what I mean. Anyway, this one may sound a bit too informal, so I proposed a way to reword it.

File: docs/features/spreadsheets/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jacekbogdanski) [reply docs/features/spreadsheets/README.md]: I seems easy when you thinking about it but then writing it down... :smile: Maybe let's just add some sentence like:

`You can even drag and drop conditional formatting rules changing its priority to easily create features like a color scale. Put your hands on the color scale example with our demo at <link to demo>`?

**Comment by jacekbogdanski** [comment]
I was rather thinking about somethig like:

* adding and removing columns / rows
* auto filling - https://handsontable.com/docs/7.2.1/demo-auto-fill.html
* resizing rows and columns - https://handsontable.com/docs/7.2.1/demo-resizing.html (we have disabled this feature but i think it could be enabled eventually as a dialog option / config option
* custom headers ( although its more a hidden feature as it only works if you create a downcasted table with custom header)

but copying visual selection also looks nice :+1: Also note that widget is downcasted when copying, so maybe something like copying and pasting Spreadsheet as a plain table?

**Comment by jacekbogdanski** [comment]
I added workaround for Spreadsheet jiggering after sample load here: https://github.com/ckeditor/ckeditor4-docs/pull/265/commits/b8bbfe18cc4b9c498654184d75d98425ee97f2e5#diff-502360ecb612581e5e2821e1ea0439aaR257-R267

This workaround should be removed once we merge https://github.com/cksource/ckeditor4-plugin-spreadsheet/pull/314 

Hopefully, fix will be available before release, so please make sure that Spreadsheet dist files are up to date and remove workaround after https://github.com/cksource/ckeditor4-plugin-spreadsheet/pull/314

(I will do it myself if the timing will be right)

**Comment by AnnaTomanek** [comment]
Looks good to go. Great job, everyone! 🎉

**Comment by jacekbogdanski** [comment]
Looks great! Let's wait for the latest Spreadsheet plugin version, so we can update the package and merge this pretty!

**Comment by f1ames** [comment]
Rebased onto latest `stable`.

**Pull request merged**